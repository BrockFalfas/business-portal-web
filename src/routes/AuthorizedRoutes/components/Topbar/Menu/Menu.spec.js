import React from 'react';
import { shallow, mount } from 'enzyme';

import { Menu, generateMenuItems } from './index';

const mockMenu = [
  { route: '/example1', label: 'First' },
  { route: '/example2', label: 'Second' },
  { route: '/example3', label: 'Third' },
];

const initMenu = overrides => {
  const mockProps = {
    pathname: '/welcome',
  };
  const wrapper = shallow(<Menu {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('component: <Menu/>', () => {
  describe('renderManageAccount', () => {
    it('should render without crashes', () => {
      const { wrapper } = initMenu();
      expect(wrapper).toBeTruthy();
    });
  });

  describe('should highlight menu item', () => {
    it('should highlight selected items based on pathname', () => {
      const { wrapper } = initMenu();
      expect(wrapper.prop('selectedKeys')).toEqual(['/welcome']);
      expect(wrapper).toMatchSnapshot();

      wrapper.setProps({ pathname: '/payments' });
      expect(wrapper.prop('selectedKeys')).toEqual(['/payments']);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render menu', () => {
      const renderedMenu = generateMenuItems(mockMenu);
      expect(renderedMenu).toBeTruthy();
    });

    it('should have three navigation items', () => {
      const renderedMenu = generateMenuItems(mockMenu);
      expect(renderedMenu).toHaveLength(3);
    });

    it('should render correct', () => {
      const renderedMenu = generateMenuItems(mockMenu);
      expect(renderedMenu).toMatchSnapshot();
    });
  });
});
