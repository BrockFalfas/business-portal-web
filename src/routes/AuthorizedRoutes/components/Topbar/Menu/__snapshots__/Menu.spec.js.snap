// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <Menu/> should highlight menu item should highlight selected items based on pathname 1`] = `
<Menu
  className="Menu"
  mode="horizontal"
  selectedKeys={
    Array [
      "/welcome",
    ]
  }
  theme="light"
>
  <MenuItem
    className="Menu__item"
    key="/welcome"
  >
    <Link
      className="Menu__link"
      replace={false}
      to="/welcome"
    >
      <span>
        Welcome
      </span>
    </Link>
  </MenuItem>
  <MenuItem
    className="Menu__item"
    key="/payments"
  >
    <Link
      className="Menu__link"
      replace={false}
      to="/payments"
    >
      <span>
        Payments
      </span>
    </Link>
  </MenuItem>
  <MenuItem
    className="Menu__item"
    key="/contractors"
  >
    <Link
      className="Menu__link"
      replace={false}
      to="/contractors"
    >
      <span>
        Contractors
      </span>
    </Link>
  </MenuItem>
</Menu>
`;

exports[`component: <Menu/> should highlight menu item should highlight selected items based on pathname 2`] = `
<Menu
  className="Menu"
  mode="horizontal"
  selectedKeys={
    Array [
      "/payments",
    ]
  }
  theme="light"
>
  <MenuItem
    className="Menu__item"
    key="/welcome"
  >
    <Link
      className="Menu__link"
      replace={false}
      to="/welcome"
    >
      <span>
        Welcome
      </span>
    </Link>
  </MenuItem>
  <MenuItem
    className="Menu__item"
    key="/payments"
  >
    <Link
      className="Menu__link"
      replace={false}
      to="/payments"
    >
      <span>
        Payments
      </span>
    </Link>
  </MenuItem>
  <MenuItem
    className="Menu__item"
    key="/contractors"
  >
    <Link
      className="Menu__link"
      replace={false}
      to="/contractors"
    >
      <span>
        Contractors
      </span>
    </Link>
  </MenuItem>
</Menu>
`;

exports[`component: <Menu/> should highlight menu item should render correct 1`] = `
Array [
  <MenuItem
    className="Menu__item"
>
    <Link
        className="Menu__link"
        replace={false}
        to="/example1"
    >
        <span>
            First
        </span>
    </Link>
</MenuItem>,
  <MenuItem
    className="Menu__item"
>
    <Link
        className="Menu__link"
        replace={false}
        to="/example2"
    >
        <span>
            Second
        </span>
    </Link>
</MenuItem>,
  <MenuItem
    className="Menu__item"
>
    <Link
        className="Menu__link"
        replace={false}
        to="/example3"
    >
        <span>
            Third
        </span>
    </Link>
</MenuItem>,
]
`;
